File: backend/app/schemas/ai_insight.py
from pydantic import BaseModel

from app.schemas.attendance_history import AttendanceCounts, SubjectAttendanceSummary


class AIInsightResponse(BaseModel):
    student_id: int
    overall: AttendanceCounts
    by_subject: list[SubjectAttendanceSummary]
    ai_available: bool
    insight_text: str | None
    ai_error_code: str | None

File: frontend/src/api/aiInsight.ts
import { apiFetch } from './client'
import type { AttendanceCounts, SubjectAttendanceSummary } from './dashboard'

export interface AIInsight {
  student_id: number
  overall: AttendanceCounts
  by_subject: SubjectAttendanceSummary[]
  ai_available: boolean
  insight_text: string | null
  ai_error_code: string | null
}

export const aiInsightApi = {
  get: (studentId: number) => apiFetch<AIInsight>(`/api/v1/students/${studentId}/ai-insight`),
}
